resources/environments/data_source_aws_credential_prerequisites.go

```diff
@@ -86,14 +86,15 @@ func (d *awsCredentialPrerequisitesDataSource) Read(ctx context.Context, req dat
 	data.ExternalID = types.StringValue(*prerequisites.Aws.ExternalID)
 	data.ID = types.StringValue(prerequisites.AccountID + ":" + *prerequisites.Aws.ExternalID)
 	data.Policy = types.StringPointerValue(prerequisites.Aws.PolicyJSON)
-	data.Policies = make([]*credentialGranularPolicyDataSourceModel, len(prerequisites.Aws.Policies))
-	for i, policy := range prerequisites.Aws.Policies {
-		data.Policies[i] = &credentialGranularPolicyDataSourceModel{
-			Service:    types.StringPointerValue(policy.Service),
-			PolicyJson: types.StringPointerValue(policy.PolicyJSON),
-		}
+
+	var policyMap = make(map[string]string)
+	for _, policy := range prerequisites.Aws.Policies {
+		policyMap[*policy.Service] = *policy.PolicyJSON
 	}
 
+	policyTypeMap, _ := types.MapValueFrom(ctx, types.StringType, policyMap)
```

Comment:
Append the diag return value to the response diagnostics to avoid ignoring conversion errors.

Reply:
updated